generator_process/future.py

```diff
@@ -64,7 +64,7 @@ def cancel(self):
     def _run_on_main_thread(self, func):
         try:
             import bpy
-            bpy.app.timers.register(func)
+            bpy.app.timers.register(func, persistent=True)
```

Comment:
Makes sure that model fetching completes when loading a file. Often when loading a file first it would fetch the models, then it gets queued in this timer, then the file actually loads and discards the timer. On rare occasions model fetching would complete along with the timer before the .blend file loads.